### ISSUE TITLE
Feature request: TITLE

### ISSUE THREAD

**Issue by adrienarcuri** [aws/aws-sam-cli-app-templates#425]
### Describe your idea/feature/enhancement

I which AWS SAM CLI Application Templates would have an explanation of every template that is presented when using the AWS SAM CLI command 'sam init'.

The template are the following:
1 - Hello World Example
2 - Data processing
3 - Hello World Example With Powertools
...
14 - Machine Learning

### Proposal

For each template, it have:
- a brief description
- a list of deployed components in AWS
- a diagram of deployed component in AWS

**Comment by mildaniel** [comment]
Hey @adrienarcuri, I like this idea. I think some of the names here might be a little bit confusing and this could be a useful UX improvement. If you're open to contributing, we would be happy to review your PR!

**Comment by adrienarcuri** [comment]
@mildaniel I'm open to contributing. 
Know that I am new user of SAM service.
If I'm not mistaken, each SAM template corresponds to a CloudFormation template, is that correct?
Could you tell me where I can find the CloudFormation templates corresponding to each SAM template please? Thanks to this information I can understand which resources are deployed for each SAM template and provide a description and a diagram for each template. Thanks in advance.

**Comment by mildaniel** [comment]
You're right that SAM templates correspond to CFN templates. SAM templates are actually supersets of CFN templates, so any CFN resource in the SAM template works as is. In terms of SAM resources, they get broken down into CFN resources, but I think it's sufficient to just use the SAM resource in the description.
